src/database/Database.js

```diff
@@ -8,47 +8,42 @@ const database_version = '1.0';
 const database_displayname = 'SQLite React Offline Database';
 const database_size = 10000000;
 
-export class UserLocationsDatabase {
-  initDB() {
-    let db;
-    return new Promise((resolve, reject) => {
-      SQLite.echoTest()
-        .then(() => {
-          SQLite.openDatabase(
-            database_name,
-            database_version,
-            database_displayname,
-            database_size
-          )
-            .then((DB) => {
-              db = DB;
-              db.executeSql('CREATE TABLE IF NOT EXISTS Samples (lat,long,accuracy,startTime,endTime,geoHash,wifiHash,hash);').then(() => {
-                resolve(db);
-              }).catch((error) => {
-                reject(error);
-              });
-            })
-            .catch((error) => {
-              console.log(error);
-              reject(error);
-            });
-        })
-        .catch((error) => {
-          console.log('echoTest failed - plugin not functional');
-          reject(error);
-        });
-    });
+class Database {
+  async initDB(initializationStatement) {
+    try {
+      await SQLite.echoTest()
+      const db = await SQLite.openDatabase(
+        database_name,
+        database_version,
+        database_displayname,
+        database_size
+      )
+      await db.executeSql(initializationStatement)
+      return db
+    }
+    catch (error) {
+      console.log('echoTest failed - plugin not functional');
+      throw error;
+    };
   }
 
   closeDatabase(db) {
     if (db) {
       db.close()
-        .catch((error) => {
+        .catch(error => {
           // this.errorCB(error);
           // TODO makes unhandled promise reject in addSample function - need to check why
-        });
+          console.log('Closing the database failed', error)
+        })
     }
   }
+}
+
+
+export class UserLocationsDatabase extends Database {
+  initDB() {
+    return super.initDB('CREATE TABLE IF NOT EXISTS Samples (lat,long,accuracy,startTime,endTime,geoHash,wifiHash,hash);')
```

Comment:
maybe, append static create (CREATE TABLE IF NOT EXISTS ) to the specific table in the base class instead of repeating it.